src/main/kotlin/com/getkeepsafe/dexcount/ColorConsole.kt

```diff
@@ -0,0 +1,201 @@
+        val sto = if (level == null) {
+            StyledTextOutputFactory_create(factory, label)
+        } else {
+            StyledTextOutputFactory_createWithLevel(factory, label, level)
+        }
+        return StyledTextOutputWrapper(sto)
+    }
+}
+
+private class StyledTextOutputWrapper(private var sto: Any) {
+    fun withStyle(style: Style): StyledTextOutputWrapper {
+        val platformStyle = style.toPlatformStyle()
+        this.sto = StyledTextOutput_withStyle(sto, platformStyle)
+        return this
+    }
+
+    fun asPrintWriter(): PrintWriter {
+        val appendable = sto as Appendable
+        val writer = AppendablePrintWriterAdapter(appendable)
+        return PrintWriter(writer)
+    }
+}
+
+private fun Style.toPlatformStyle(): Any {
+    val name = this.name
+    return styleClass.enumConstants.single { it.name == name }
+}
+
+private class AppendablePrintWriterAdapter(private val appendable: Appendable) : Writer() {
+    override fun append(csq: CharSequence?): Writer {
+        appendable.append(csq)
+        return this
+    }
+
+    override fun append(csq: CharSequence?, start: Int, end: Int): Writer {
+        appendable.append(csq, start, end)
+        return this
+    }
+
+    override fun append(c: Char): Writer {
+        appendable.append(c)
+        return this
+    }
+
+    override fun write(c: Int) {
+        super.write(c)
+    }
+
+    override fun write(cbuf: CharArray?) {
+
+    }
+
+    override fun write(cbuf: CharArray?, off: Int, len: Int) {
+        val str = String(cbuf!!, off, len)
+        appendable.append(str)
+    }
+
+    override fun write(str: String?) {
+        appendable.append(str)
+    }
+
+    override fun write(str: String?, off: Int, len: Int) {
+        val substr = str?.substring(off, off + len)
+        appendable.append(substr)
+    }
+
+    override fun flush() {
+    }
+
+    override fun close() {
+    }
+}
+
+private val styledTextOutputFactoryClass: Class<*> by lazy {
+    getClassWithFallback(
+        "org.gradle.internal.logging.text.StyledTextOutputFactory",
+        "org.gradle.logging.StyledTextOutputFactory"
+    )
+}
+
+private val styledTextOutputClass: Class<*> by lazy {
+    getClassWithFallback(
+        "org.gradle.internal.logging.text.StyledTextOutput",
+        "org.gradle.logging.StyledTextOutput"
+    )
+}
+
+@Suppress("UNCHECKED_CAST")
+private val styleClass: Class<Enum<*>> by lazy {
+    getClassWithFallback(
+        "org.gradle.internal.logging.text.StyledTextOutput\$Style",
+        "org.gradle.logging.StyledTextOutput\$Style"
+    ) as Class<Enum<*>>
+}
+
+private val serviceRegistryClass: Class<*> by lazy {
+    getClassWithFallback(
+        "org.gradle.internal.service.ServiceRegistry",
+        "org.gradle.api.internal.project.ServiceRegistry"
+    )
+}
+
+private val AbstractTask_getServices: Method by lazy {
+    var clazz: Class<*>? = DefaultTask::class.java
```

Comment:
All this crap is due to the fact that AbstractTask is internal.  All we can rely on is that DefaultTask extends from _something_ that has a protected "getServices" method.